### PR TITLE
feat GAT-40: created BlogHead block and BlogHead component.

### PR DESCRIPTION
**Pull request by melscastillo**
Ticket: GAT-40 https://app.clickup.com/t/2257946/GAT-40
CHANGES: We created the BlogHead component and the BlogHead block.

Storybook:
- Block:
image
- Component:
image

### CODE REVIEW

File: packages/gatsby-theme/src/blocks/BlogHeadBlock/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import PropTypes from "prop-types"
+import BlogHead from "../../components/BlogHead"
```
Comment (luisduenas) [inline packages/gatsby-theme/src/blocks/BlogHeadBlock/index.js]: Please add BlogHead component to the [component index file](https://github.com/octahedroid/drupal-gatsby-theme/blob/master/packages/gatsby-theme/src/components/index.js) and update this import
```suggestion
import { BlogHead } from "ui"
```

File: packages/gatsby-theme/src/blocks/BlogHeadBlock/BlogHeadBlock.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  backgroundColor: {
+    description: "Background overlay color",
+    control: {
+      type: "color",
+      presetColors: ["#001133CC"],
+    },
+  },
+}
+
+<Meta title='blocks/BlogHeadBlock' component={BlogHeadBlock} />
```
Comment (dinartecc) [inline packages/gatsby-theme/src/blocks/BlogHeadBlock/BlogHeadBlock.stories.mdx]: We should pass the `argTypes` object directly into this Meta tag. This applies it to all the stories in this file.

File: packages/gatsby-theme/src/blocks/BlogHeadBlock/BlogHeadBlock.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+## Props
+
+<Props of={BlogHeadBlock} />
+
+export const BlogHeadBlockTemplate = (args) => <BlogHeadBlock {...args} />
+
+### Default BlogHeadBlock
+
+<Preview>
+  <Story name='Default' args={{ ...data }} argTypes={blogHeadArgTypes}>
```
Comment (dinartecc) [inline packages/gatsby-theme/src/blocks/BlogHeadBlock/BlogHeadBlock.stories.mdx]: We should remove the argTypes from all the stories in favor of adding it in the `Meta` component above.

File: packages/gatsby-theme/src/components/BlogHead/index.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import BlogHeadCredits from "./BlogHeadCredits"
+
+BlogHead.Image = BlogHeadImage
+BlogHead.Title = BlogHeadTitle
+BlogHead.Author = BlogHeadAuthor
+BlogHead.Date = BlogHeadDate
+BlogHead.Container = BlogHeadContainer
+BlogHead.Credits = BlogHeadCredits
+
+export default BlogHead
```
Comment (dinartecc) [inline packages/gatsby-theme/src/components/BlogHead/index.js:16]: Please add this component to the `ui` index under `frontend/packages/gatsby-theme-octahedroid/index.js`

**Review by luisduenas — APPROVED** [review]
LGTM

**Pull request merged**